Library/Homebrew/extend/ENV/shared.rb

```diff
@@ -63,7 +63,7 @@ def reset
   sig { returns(T::Hash[String, String]) }
   def remove_cc_etc
     keys = %w[CC CXX OBJC OBJCXX LD CPP CFLAGS CXXFLAGS OBJCFLAGS OBJCXXFLAGS LDFLAGS CPPFLAGS]
-    keys.to_h { |key| [key, delete(key)] }
+    keys.to_h { |key| [key, T.must(delete(key))] }
```

Comment:
Is this correct? I think this should either return `T::Hash[String, T.nilable(String)]` or we need to call `Hash#compact` before returning it.

The only use where the return value is used is when passing it to `ENV.update`, so I assume it should be the former so that the variables are cleared which weren't set before.